### DISCUSSION TITLE
Resources using property create pattern

### DISCUSSION THREAD

**Discussion by SimonCropp** [microsoft/aspire#5487] [General]
i notice quite a few of the immutable *resource classes are using the "property create instance on get" approach.

so creating a new instance every time the property is accessed

```
public BicepOutputReference ConnectionString => new("connectionString", this);
```

I am just wondering if this is intentional? ie instead of placing it in a backing field

![image](https://github.com/user-attachments/assets/138fbcd9-3844-40ae-9866-3d1084b20276)

**Comment by davidfowl** [comment]
These could and probably should be backing fields. I’d suggest not sending a mega PR though, those types of refactoring are disruptive for little benefit. We can fix these incrementally as we change the classes.
